samcli/commands/_utils/options.py

```diff
@@ -35,15 +35,15 @@ def get_or_default_template_file_name(ctx, param, provided_value, include_build)
 
     original_template_path = os.path.abspath(provided_value)
 
-    search_paths = ["template.yaml", "template.yml"]
+    search_paths = ["template.yaml", "template.yml", "template.json"]
```

Comment:
something to check, if build is included, we always the convert the built template back to yaml (IIRC) and that is referenced template if one is not provided during the deploy command, can we check this behavior?

Reply:
> something to check, if build is included, we always the convert the built template back to yaml (IIRC)

If `sam build` is executed, `template.json` will be rewritten into `.aws-sam/build/template.yaml`. And if `sam package/deploy` is later executed, `.aws-sam/build/template.yaml` should be picked up. Nice to double check this behavior. 

@sriram-mv so we want the priority in `sam local` and `sam package/deploy` to be:
1. .aws-sam/build/template.yaml
2. ./template.yaml
3. ./template.yml
4. ./template.json
is that accurate?

Reply:
@sriram-mv Yes, when we run build, `template.json` is converted to `template.yaml` in `.aws-sam/build` after `sam build` execution. It works without explicit `--template` parameter

Reply:
ok sounds good, it (template.json) should be the last thing to look up in order to avoid any breaking changes or unwanted behavior from this.

Reply:
Something I just realised. the parser is still a yaml_parser.

yaml_helper.py

```
def yaml_parse(yamlstr):
    """Parse a yaml string"""
    try:
        # PyYAML doesn't support json as well as it should, so if the input
        # is actually just json it is better to parse it with the standard
        # json parser.
        return json.loads(yamlstr, object_pairs_hook=OrderedDict)
    except ValueError:
        yaml.SafeLoader.add_constructor(yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG, _dict_constructor)
        yaml.SafeLoader.add_multi_constructor("!", intrinsics_multi_constructor)
        return yaml.safe_load(yamlstr)
```

if you look at the comment, it suggest use a json parser instead.